### PR TITLE
LPS-196698 Change the message and show the limit of the preview-able files

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
image

- LPS-181593 change message
- LPS-196698 add method to the processor registry to provide the max value of the previweable processor from the util
- LPS-196698 baseline
- LPS-196698 change exception to show the max size allowed for the preview
- LPS-196698 baseline
- LPS-196698 change message to show the new message and the maz size allowed
- LPS-196698 buildLang

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -18608,6 +18607,7 @@ this-field-requires-a-valid-style-value=This field requires a valid style value.
 this-field-type-does-not-support-translations=This field type does not support translations.
 this-file-contains-headers=This File Contains Headers
 this-file-extension-is-not-supported-by-docusign=This file extension is not supported by DocuSign.
+this-file-is-too-large-to-preview.-the-maximum-size-for-preview-files-is-x=This file is too large to preview. The maximum size for preview files is {0}.
```
Comment (marcogalluzzi) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18610]: You know I'm not a native English speaker but I would say that the sentence is not correct. As I understand the max file size is for the original file: when the original file is less or equal to this max file size then a preview is generated. Therefore, the last part of the sentence is not correct:

* _The maximum size for preview files is X_ means that "preview files" have a maximum size of X (not true)
* This can be fixed by replacing "for preview files" with "for previewing files" or "to preview files".

What do you think @austinchiang ?

File: modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java [hunk 2]
```diff
@@ -13,18 +14,53 @@
 public class DLPreviewSizeException extends PortalException {
 
 	public DLPreviewSizeException() {
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java:16]: I would remove constructors without `maxSize`, see comment below.

File: modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java [hunk 3]
```diff
@@ -13,18 +14,53 @@
...
+		this(0);
+	}
+
+	public DLPreviewSizeException(long maxSize) {
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java:20]: Can you please call it `maxFileSize`? I can foresee the future: the configuration attribute will be called this way :-D

File: modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java [hunk 4]
```diff
@@ -13,18 +14,53 @@
...
+		super(throwable);
+
+		_maxSize = maxSize;
 	}
 
 	public DLPreviewSizeException(String msg) {
+		this(msg, 0);
+	}
+
+	public DLPreviewSizeException(String msg, long maxSize) {
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java:34]: Parameters need sorting?

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18610]: Well, I'm awful on english too so I tend to copy what is put on the story/figma :D

Please, if is not correct, I will change it.

File: modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java (hunk 3 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java:20]: I will read the future and call it this way :P

File: modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java (hunk 4 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java:34]: on this case, the msg of the exception always goes fist.
golden rule on my mind

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18610]: I'm not sure about this :( 

The preview size is not involved here.
Is the size of the file we will make the preview...


This modify anything?

File: modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java [hunk 5]
```diff
@@ -13,18 +14,53 @@
...
+		this(0, throwable);
 	}
 
+	public long getMaxSize() {
+		if (_maxSize != 0) {
+			return _maxSize;
+		}
+
+		return DLProcessorRegistryUtil.getPreviewableProcessorMaxSize();
+	}
```
[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java:62]: I'm not sure about this.
I do not think that make the size a needed parameter to create an exception (any exception, a needed parameter) is a good practice.

So if you are sure about this, can we make a call about it.

File: modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/preview/exception/DLPreviewSizeException.java:15]: discarded

**Review by AliciaGarciaGarcia — COMMENTED** [review]
@adolfopa 

I tried to follow the new "Process to register breaking changes in commits"
Can you review this commit message?  :D

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 1 above)
Comment (austinchiang) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties]: I think there is enough context from the previous sentence ("This file is too large to preview") to infer that the maximum size is referring to the size of the file. But if we want to make it more explicit then maybe something like "The maximum file size for previews is {0}"

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp [hunk 6]
```diff
@@ -23,7 +23,16 @@ FileVersion fileVersion = (FileVersion)request.getAttribute(WebKeys.DOCUMENT_LIB
...
 			<p class="text-secondary">
-				<liferay-ui:message arguments="<%= LanguageUtil.formatStorageSize(dlPreviewSizeException.getMaxFileSize(), locale) %>" key="this-file-is-too-large-to-preview.-the-maximum-size-for-preview-files-is-x" />
+
+				<%
+				long maxFileSize = dlPreviewSizeException.getMaxFileSize();
+
+				if (maxFileSize == 0) {
+					maxFileSize = DLProcessorRegistryUtil.getPreviewableProcessorMaxSize();
+				}
+				%>
```
Comment (marcogalluzzi) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp:33]: Why did you move this logic here instead of passing the `DLProcessorRegistryUtil.getPreviewableProcessorMaxSize()` to the constructor of the Exception, this is in files `*PreviewRendererProvider`? If you prefer this way then you can simply remove the `maxFileSize` attribute from the `DLPreviewSizeException` class, it becomes useless.

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia There's an inconsistency in the documentation of the message format (in one place it shows "What:" -- with a colon -- and in the example the colon is missing, so I don't know which one is the correct one. Other than that, it looks fine. I've already reported it, hopefully it will be corrected today.

Other than that, everything looks fine.

**Review by adolfopa — APPROVED** [review]

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp (hunk 6 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp:33]: Also, take into account that the configuration attribute `previewableProcessorMaxSize` will be configurable at group level. The group level can be easily taken from the `fileVersion` in `*PreviewRendererProvider` files. Here I suppose it's harder to know for which `fileVersion` was the exception thrown.

Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp:33]: > Why did you move this logic here instead of passing the `DLProcessorRegistryUtil.getPreviewableProcessorMaxSize()` to the constructor of the Exception, this is in files `*PreviewRendererProvider`? If you prefer this way then you can simply remove the `maxFileSize` attribute from the `DLPreviewSizeException` class, it becomes useless.

We cannot assume that the exception may have the field with data, so we must have a way to display the value if it does not come.
And to say "if not comes it's a developer problem" it's not good for me. So I prefer to prevent the misuse of the exception that to show bad information.

Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp:33]: > Also, take into account that the configuration attribute `previewableProcessorMaxSize` will be configurable at group level. The group level can be easily taken from the `fileVersion` in `*PreviewRendererProvider` files. Here I suppose it's harder to know for which `fileVersion` was the exception thrown.

But now we do not have this field so I can not make the method right now with the fileVersion.
And YES we know which fileversion is failing (see the jsp, has fileVersion.getFileEntry()) so in the future we can pass the group of whatever we need. :)

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties]: thanks @austinchiang

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp (hunk 6 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp:33]: Then, please remove the `maxFileSize` attribute from the Exception class. It's now useless.

Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp:33]: As we talked yesterday, when the PRs about the videos, images, audios and documents unification (now on review)
we will pass the value on the exception for my changes.
But we can not prevent future developers that will call a simply `new DLPreviewSizeException()` and I want to prevent the system present the user wrong information

Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp:33]: I didn't get that yesterday, sorry.

**Review by marcogalluzzi — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
changed de message with the @austinchiang suggestion.

Thanks!

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
ci:test:sf

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
ci:test:document

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
Hi @AliciaGarciaGarcia Could you help to confirm if the below failure is related to this pr? Thanks.

[modules-semantic-versioning-jdk8/0/0](https://testray.liferay.com/home/-/testray/case_results/2430242312?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2430239356%26testrayRunId%3D2430239397%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

```
 [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/document-library/document-library-api] STARTED
     [exec] 
     [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/document-library/document-library-api] FAILED
     [exec]     java.lang.AssertionError: [Baseline Warning] Bundle Version Change Recommended: 23.3.0
     [exec]         at org.junit.Assert.fail(Assert.java:88)
     [exec]         at org.junit.Assert.assertTrue(Assert.java:41)
     [exec]         at org.junit.Assert.assertFalse(Assert.java:64)
     [exec]         at com.liferay.semantic.versioning.SemanticVersioningTest.testSemanticVersioning(SemanticVersioningTest.java:73)
     [exec]         at sun.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
     [exec]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]         at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]         at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
     [exec]         at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
     [exec]         at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
     [exec]         at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
     [exec]         at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
     [exec]         at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
     [exec]         at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
     [exec]         at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
     [exec]         at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
     [exec]         at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
     [exec]         at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
     [exec]         at org.junit.runners.Suite.runChild(Suite.java:128)
     [exec]         at org.junit.runners.Suite.runChild(Suite.java:27)
     [exec]         at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
     [exec]         at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
     [exec]         at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
     [exec]         at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
     [exec]         at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
     [exec]         at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
     [exec]         at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]         at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]         at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
     [exec]         at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
     [exec]         at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
     [exec]         at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
     [exec]         at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
     [exec]         at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
     [exec]         at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
     [exec]         at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
     [exec]         at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
     [exec]         at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
     [exec]         at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
     [exec]         at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
     [exec]         at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
     [exec]         at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

**Comment by AliciaGarciaGarcia** [comment]
Hi @Jeremy-Ch 

thanks!

With the merge on master a baseline was necessary.
I made it and a rebase

**Comment by AliciaGarciaGarcia** [comment]
c:test:sf

**Comment by Jeremy-Ch** [comment]
c:test:sf

**Comment by Jeremy-Ch** [comment]
ci:test:stable

**Comment by Jeremy-Ch** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:forward:force
